tests/Microsoft.Identity.Test.Unit/pop/PopAuthenticationSchemeTests.cs

```diff
@@ -104,60 +104,93 @@ public void ValidatePopRequestAndToken()
                 PoPAuthenticationConfiguration popConfig = new PoPAuthenticationConfiguration(new Uri("https://www.contoso.com/path1/path2?queryParam1=a&queryParam2=b"));
                 popConfig.HttpMethod = HttpMethod.Get;
-                popConfig.PopCryptoProvider = new InMemoryCryptoProvider();
 
                 var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
                                 .WithHttpManager(harness.HttpManager)
                                 .WithExperimentalFeatures()
                                 .WithClientSecret("some-secret")
                                 .BuildConcrete();
 
+                TokenCacheHelper.PopulateCache(app.AppTokenCacheInternal.Accessor);
+
+                harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(
+                    authority: TestConstants.AuthorityCommonTenant,
+                    responseMessage: MockHelpers.CreateSuccessfulClientCredentialTokenResponseMessage(token:$"header.{Guid.NewGuid()}.signature", tokenType: "pop"));
+
                 harness.HttpManager.AddSuccessTokenResponseMockHandlerForPost(
                     authority: TestConstants.AuthorityCommonTenant,
-                    responseMessage: MockHelpers.CreateSuccessfulClientCredentialTokenResponseMessage(tokenType: "pop"));
+                    responseMessage: MockHelpers.CreateSuccessfulClientCredentialTokenResponseMessage(token: $"header.{Guid.NewGuid()}.signature", tokenType: "pop"));
 
                 Guid correlationId = Guid.NewGuid();
                 TestClock testClock = new TestClock();
                 testClock.TestTime = DateTime.UtcNow;
-                var provider = PoPProviderFactory.GetOrCreateProvider(testClock);
+                PoPProviderFactory.TimeService = testClock;
+                var provider = PoPProviderFactory.GetOrCreateProvider();
 
-                await app.AcquireTokenForClient(TestConstants.s_scope)
+                var result = await app.AcquireTokenForClient(TestConstants.s_scope)
                     .WithProofOfPossession(popConfig)
                     .ExecuteAsync(CancellationToken.None)
                     .ConfigureAwait(false);
+                var initialToken = result.AccessToken;
 
                 var JWK = provider.CannonicalPublicKeyJwk;
-                //Advance time 7 hours. Should still be the same key
-                testClock.TestTime = testClock.TestTime + TimeSpan.FromSeconds(60 * 60 * 7);
 
-                provider = PoPProviderFactory.GetOrCreateProvider(testClock);
-                await app.AcquireTokenForClient(TestConstants.s_scope)
+                //Advance time 7 hours. Should still be the same key and token
+                testClock.TestTime = testClock.TestTime + TimeSpan.FromHours(7);
+
+                PoPProviderFactory.TimeService = testClock;
+                provider = PoPProviderFactory.GetOrCreateProvider();
```

Comment:
Can you remove this from the test please? To make sure there aren't any side effects. It's ok not to assert the JWK.